### PR TITLE
Add prune option

### LINKED ISSUE #47: Don't delete unmanaged labels
In some cases it will be appreciated to don't remove unmanaged labels.

eg. dependabot adds some label to manage dependencies - I want leave untouched such labels.

In current situation I must managed labels added by other tools.

So option for this case will be good.

### PR DESCRIPTION
**Pull request by slawekjaranowski**
default behavior was preserved

fix #47

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -12,6 +12,10 @@ inputs:
   token:
     description: "An alternative GitHub token to use instead"
     required: false
+  deleteUnManaged:
```
Comment (micnncim) [inline action.yml:15]: I prefer `prune` to `deleteUnManaged` here (in any other place as well).

File: cmd/action-label-syncer/main.go [hunk 2]
```diff
@@ -48,8 +49,19 @@ func main() {
 	}
 	owner, repo := slugs[0], slugs[1]
 
+	deleteUnManagedStr := os.Getenv("INPUT_DELETEUNMANAGED")
+	if len(deleteUnManagedStr) == 0 {
```
Comment (micnncim) [inline cmd/action-label-syncer/main.go:53]: [nitpick] I prefer `stringVar == ""` to check if a string is empty.

File: cmd/action-label-syncer/main.go [hunk 3]
```diff
@@ -48,8 +49,19 @@ func main() {
...
+		deleteUnManagedStr = "true"
```
Comment (micnncim) [inline cmd/action-label-syncer/main.go:54]: Since the default value is `true`, you could omit this.

File: pkg/github/github.go [hunk 4]
```diff
@@ -74,19 +75,21 @@ func (c *Client) SyncLabels(ctx context.Context, owner, repo string, labels []La
...
-		currentLabel := currentLabel
-		eg.Go(func() error {
-			_, ok := labelMap[currentLabel.Name]
-			if ok {
-				return nil
-			}
-			return c.deleteLabel(ctx, owner, repo, currentLabel.Name)
-		})
-	}
+	if deleteUnManaged {
```
Comment (micnncim) [inline pkg/github/github.go:78]: Use early return.

File: pkg/github/github.go [hunk 5]
```diff
@@ -108,12 +112,14 @@ func (c *Client) SyncLabels(ctx context.Context, owner, repo string, labels []La
 }
 
 func (c *Client) createLabel(ctx context.Context, owner, repo string, label Label) error {
+
```
Comment (micnncim) [inline pkg/github/github.go:115]: This is an unnecessary empty line. Please delete it.

**Review by micnncim — CHANGES_REQUESTED** [review]
Overall LGTM though I left several comments.

One more thing; please follow the commit message convention and update your commit message and PR title like `Add prune option`.

File: cmd/action-label-syncer/main.go (hunk 2 above)
Comment (slawekjaranowski) [reply cmd/action-label-syncer/main.go:53]: all other comparison is done by `len(...)`

File: pkg/github/github.go (hunk 4 above)
Comment (slawekjaranowski) [reply pkg/github/github.go:78]: I don't know how to use early return here we need rest of this method

File: action.yml [hunk 6]
```diff
@@ -12,6 +12,10 @@ inputs:
   token:
     description: "An alternative GitHub token to use instead"
     required: false
+  prune:
+    description: "remove unmanaged labels from repository"
```
Comment (micnncim) [inline action.yml:16]: [nitpick]

```suggestion
    description: "Remove unmanaged labels from repository"
```

**Review by micnncim — COMMENTED** [review]
Perfect other than a nitpick - thanks!

**Review by micnncim — APPROVED** [review]

**Comment by micnncim** [comment]
Great work - I appreciate your contributions 😄

**Comment by micnncim** [comment]
Created the new tag `v1.2.0` and [release](https://github.com/micnncim/action-label-syncer/releases/tag/v1.2.0). Also, bound the tag with `v1`.

**Comment by slawekjaranowski** [comment]
Thx for approve and new release.

**Pull request merged**